Makefile

```diff
@@ -14,25 +14,41 @@ checkout_master:
 
 .PHONY: create_release
 .SILENT: create_release
-create_release:
+create_release: set_release_version update_release_files_commit
+
+.PHONY: create_release
+.SILENT: create_release
+set_release_version:
 	git checkout -b $(RELEASE_BRANCH) && \
```

Comment:
Do you need to merge command lines in Makefile? Doesn't it abort shell execution on first error?
https://stackoverflow.com/questions/10559064/terminate-makefile-command-when-subcommand-throws

Reply:
If you want to run more than one command per recipe then you need to do it. The overhead of that for this use case would be massive.

Reply:
This is what I am seeing with this diff:

```
-       git checkout -b $(RELEASE_BRANCH) && \
+       echo "Checking out $(RELEASE_BRANCH)"
+       git checkout -b $(RELEASE_BRANCH)
        mvn build-helper:parse-version versions:set -DgenerateBackupPoms=false \
         -DnewVersion=$$\{parsedVersion.majorVersion\}.$$\{parsedVersion.minorVersion\}.$$\{parsedVersion.incrementalVersion\}
 ```

```
make release
echo "Checking out release"
Checking out release
git checkout -b release
Switched to a new branch 'release'
mvn build-helper:parse-version versions:set -DgenerateBackupPoms=false \
	 -DnewVersion=$\{parsedVersion.majorVersion\}.$\{parsedVersion.minorVersion\}.$\{parsedVersion.incrementalVersion\}
Picked up JAVA_TOOL_OPTIONS: -Dlog4j2.formatMsgNoLookups=true
[INFO] Scanning for projects...
```

and if I try again it errors out properly given that the `release` branch already exists

```
git checkout -b release && \
	mvn build-helper:parse-version versions:set -DgenerateBackupPoms=false \
	 -DnewVersion=$\{parsedVersion.majorVersion\}.$\{parsedVersion.minorVersion\}.$\{parsedVersion.incrementalVersion\}
fatal: A branch named 'release' already exists.
make: *** [set_release_version] Error 128
```

Seems to be working fine and no need to chain everything with `&& \`?

Reply:
Right, the problem is that each recipe line is executed in a different shell, so when you use variables like with the `DATE` further down you will need to use a single shell. I tend to add them elsewhere to make it easier to edit later on, but happy to remove them where no needed.

Reply:
It's updated now, one shell only where necessary.